bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewShadeHandler.java

```diff
@@ -596,4 +611,100 @@ private void doRefreshShade(RefreshKind kind) {
             }
         }
     }
+
+    /**
+     * Initialize the dynamic channels if the respective device supports them.
+     */
+    private void updateDynamicChannels() {
+        if (!dynamicChannelsInitialized) {
+            dynamicChannelsInitialized = dynamicChannelsInitializer();
+        }
+    }
+
+    /**
+     * Helper to do the actual work of initializing the dynamic channels if the respective device supports them.
+     *
+     * @return true if all channels were successfully initialised.
+     */
+    private boolean dynamicChannelsInitializer() {
```

Comment:
Would it be possible to create a builder class for some of this logic and keep this handler a bit more clean (or at least smaller). See other builders in subdirectory "builders". As an example, see method `updateSceneChannels()` how the footprint was reduced in `HDPowerViewHubHandler` with most logic put into the builder class (which is then also covered by tests).

Reply:
Yeah.

Reply:
Unfortunately it does not move 'most of the logic' to the builder. But it is nevertheless done.